docs/disk_cleaning.md

```diff
@@ -0,0 +1,140 @@
+explain below why it is so. When set to `disabled`, cleaning will be skipped as the name
+suggests, while `metadata` enables it.
+
+Advantages:
+
+- no need to rebuild a new Ironic container image to switch on/off automated cleaning mode
+- simply edit K8S CR to switch on/off automated cleaning
+- possibility to switch on/off automated cleaning selectively per node
+
+Disadvantages:
+
+- accidental editing of a template/machine/host might introduce high risks with data corruption
+
+#### Which resources do I need to edit
+
+As mentioned above, `automatedCleaningMode` exists in template, machine
+and host CRs. If you are using only Baremetal Operator (i.e. no CAPI & CAPM3), you
+will not have template & machine objects installed in your Kubernetes
+cluster and instead only host CR. As such, you should edit host CR
+`automatedCleaningMode` field to switch on/off the cleaning for your desired hosts.
+
+If you are using CAPI and CAPM3 and top of Baremetal Operator, then you can and should
+control the cleaning from CAPM3 CR, specifically template.
+
+Although, you can edit `automatedCleaningMode` value from machine or host, we
+recommend doing it from template, because template controller always
+tries to replicate the value of `automatedCleaningMode` from the template to the machine. In
+other words, template takes precedence over machine, and machine's value for
+`automatedCleaningMode` will be overridden by the value in template.
+
+If you are wondering why `automatedCleaningMode` exists in machine too, that's because
+CAPM3 machine controller copies `automatedCleaningMode` field value from a machine to its
+corresponding host. Note that machine takes precedence over host, so if
+the value `automatedCleaningMode` doesn't match between host and machine, CAPM3 machine controller
+will edit the host. However, this is not always the case, see below for more info.
+
+#### When should I edit my CRs to disable cleaning
+
+Automated cleaning takes place when its first provisioning or during each deprovisioning of a
+host. You can disable cleaning before provisioning or deprovisioning. Disabling before
+deprovisioning means any data on the host will not be wiped out and kept as it is. Note that
+this may introduce security vulnerabilities if you have sensitive data which must be wiped out
+when the host is being recycled.
+
+#### What happens if I unset automatedCleaningMode in template
+
+`automatedCleaningMode` accepts `metadata` and `disabled` or it can be unset. Please note that,
+there is no default value in CAPM3 template and machine objects for `automatedCleaningMode`. So,
+if you don't set either metadata and disabled in the template, it will be unset both in the
+template and machine objects.
+
+**Note:** There is a default value, `metadata`, in the host CR. For that reason, even if it is
+unset in the template and machine objects, Ironic node will still face automated cleaning because
+the host has enabled it by its default value.
```

Comment:
This part is a bit unclear. So, if m3mt takes precedence over m3m and if the field is unset in m3mt, that means m3m would have `automatedCleaningMode` field also unset. And as you outlined m3m has also precedence over BMH, CAPM3 machine controller will copy value from a machine to its corresponding BMH, but if BMH has default value, should not it also overriden and made unset?

Reply:
> And as you outlined m3m has also precedence over BMH, CAPM3 machine controller will copy value from a machine 
> to its corresponding BMH, but if BMH has default value, should not it also overriden and made unset?
>

No it should not unset on the host, because then you will not be able to selectively control the cleaning mode per node.

Machine takes precedence over the host, only if `automatedCleaningMode` is not unset, in all other cases the value will be copied from machine to the host.

Reply:
No, it should not, because in that you case there is no way to selectively/individually switch on/off the cleaning. If unset, that means fall back to the default in BMH.

Machine takes precedence over a host only when `automatedCleaningMode` is unset, in all other cases machine controller ensures that machine and the host have the same config.